pkg/interactor/delete_data.go

```diff
@@ -6,7 +6,7 @@ import "fmt"
 func (d *DataInteractor) Delete(token, deviceID string) error {
 	_, err := d.things.List(token)
 	if err != nil {
-		return err
+		return ErrValidToken
```

Comment:
I'm not too sure about ignoring the error returned by `List`. Is this cannot be wrapped here?

Reply:
It'd be good to improve tracing/debugging.

Reply:
I totally agree with you, but if I don't put a static error, I can't check it later in the controller. Do you guys have any suggestions to get over this?

Just a note. The errors returned by the authentication service aren't mapped.

Reply:
I understand you @Igoraamc. If I remember well in golang it's possible to use logic operations in `case` statements. Maybe you can use this with `errors.Is` method, what do you think about it?

Reply:
I changed `errors.Is` for `strings.Contains`. Do you have any problem with that?

Reply:
@Igoraamc just out of curiosity, did you face any problems when using `errors.Is`?

Reply:
Yes, for some reason, `errors.Is` wasn't detecting the error inside.